gateway-sp-comms/src/single_sp.rs

```diff
@@ -240,6 +327,18 @@ impl SingleSp {
         Ok(data)
     }
 
+    /// Abort an in-progress update.
+    pub async fn update_abort(&self, component: SpComponent) -> Result<()> {
```

Comment:
A random thought here. It's possible an abort gets delayed in the network and comes in after a new stream starts. I don't think this is a major problem, but it may be worth considering in a follow up if we want to request the current stream-id from the aborting MGS, and then cancel using that so if we start a new one we don't have this issue.